build/PackageArchive.targets

```diff
@@ -4,48 +4,68 @@
     <_WorkRoot>$(RepositoryRoot).w\</_WorkRoot>
     <MetapackageRestoreSource>$(RepositoryRoot).deps\Signed\Packages\</MetapackageRestoreSource>
     <LzmaOutputPackageName>nuGetPackagesArchive-$(PackageVersion)</LzmaOutputPackageName>
+    <FallbackStagingDir>$(_WorkRoot)obj\$(LzmaOutputPackageName)</FallbackStagingDir>
+    <FallbackOutputDir>$(ArtifactsDir)lzma\</FallbackOutputDir>
+    <FallbackOutputPath>$(FallbackOutputDir)$(LzmaOutputPackageName).lzma</FallbackOutputPath>
   </PropertyGroup>
 
-  <Target Name="BuildFallbackArchive" DependsOnTargets="ResolveRepoInfo">
-    <!-- Clear the directories -->
-    <RemoveDir Directories="$(_WorkRoot)" />
+  <Target Name="_RestoreFallbackArchive" DependsOnTargets="ResolveRepoInfo">
+    <PropertyGroup>
+      <GeneratedFallbackRestoreSourcesPropsPath>$(_WorkRoot)restoresources.$(LzmaOutputPackageName).props</GeneratedFallbackRestoreSourcesPropsPath>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_FallbackArchiveRestoreSources Include="$(RestoreSources)" />
+      <_FallbackArchiveRestoreSources Include="$(BuildDir)" Condition="Exists($(BuildDir))" />
+      <_FallbackArchiveRestoreSources Include="$(MetapackageRestoreSource)" Condition="Exists($(MetapackageRestoreSource))" />
+
+      <_MetapackageReferenceType Include="AppMetapackage;AllMetapackage"/>
+    </ItemGroup>
 
     <!-- Copy the archive template -->
     <Copy SourceFiles="$(_TemplatesDirectory)Archive\Archive.csproj" DestinationFiles="$(_WorkRoot)Archive.csproj" />
 
-    <!-- Copy the archive template -->
+    <!-- Add LZMA references -->
     <RepoTasks.AddArchiveReferences
       ReferencePackagePath="$(_WorkRoot)Archive.csproj"
+      TargetFramework="$(ArchiveTargetFramework)"
       BuildArtifacts="@(ArtifactInfo)"
       PackageArtifacts="@(PackageArtifact)"
       ExternalDependencies="@(ExternalDependency)"
       MetapackageVersion="$(PackageVersion)" />
 
-    <PropertyGroup>
-      <FallbackStagingDir>$(_WorkRoot)obj\$(LzmaOutputPackageName)</FallbackStagingDir>
-      <FallbackOutputDir>$(ArtifactsDir)lzma\</FallbackOutputDir>
-      <FallbackOutputPath>$(FallbackOutputDir)$(LzmaOutputPackageName).lzma</FallbackOutputPath>
-      <GeneratedFallbackRestoreSourcesPropsPath>$(_WorkRoot)restoresources.$(LzmaOutputPackageName).props</GeneratedFallbackRestoreSourcesPropsPath>
-    </PropertyGroup>
-
-    <ItemGroup>
-      <_FallbackArchiveRestoreSources Include="$(RestoreSources)" />
-      <_FallbackArchiveRestoreSources Include="$(BuildDir)" Condition="Exists($(BuildDir))" />
-      <_FallbackArchiveRestoreSources Include="$(MetapackageRestoreSource)" Condition="Exists($(MetapackageRestoreSource))" />
-    </ItemGroup>
+    <!-- Add metapackage references to project -->
+    <RepoTasks.AddMetapackageReferences
```

Comment:
Lots of redundant references but this guarantees we have everything for both netstandard2.0 and netcoreapp2.1.